### PR TITLE
WIP! add a silent option to GroupChatManager to reduce noise on the command line

### LINKED ISSUE #1335: [Feature Request]: allow silent=True while initiating a group chat
### Is your feature request related to a problem? Please describe.

group_chat always print to console, there is no way to prevent the print

### Describe the solution you'd like

allow silent=True in initiate_chat()

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jtoy**
## Why are these changes needed?

This will make management on the CLI easier. It is not done yet.

## Related issue number

Closes #1335

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/groupchat.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ def custom_speaker_selection_func(
     speaker_transitions_type: Literal["allowed", "disallowed", None] = None
     enable_clear_history: Optional[bool] = False
     send_introductions: bool = False
+    silent: bool = False
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:77]: not used

File: autogen/agentchat/groupchat.py [hunk 2]
```diff
@@ -516,6 +517,7 @@ def __init__(
         max_consecutive_auto_reply: Optional[int] = sys.maxsize,
         human_input_mode: Optional[str] = "NEVER",
         system_message: Optional[Union[str, List]] = "Group chat manager.",
+        silent: Optional[bool] = False,
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:520]: you need to add an object variable `self._silent = silent` in the init for it to be useable in the rest of the class

File: autogen/agentchat/groupchat.py [hunk 3]
```diff
@@ -596,7 +598,7 @@ def run_chat(
             # Broadcast the intro
             intro = groupchat.introductions_msg()
             for agent in groupchat.agents:
-                self.send(intro, agent, request_reply=False, silent=True)
+                self.send(intro, agent, request_reply=False, silent=self.silent)
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:601]:
```suggestion
                self.send(intro, agent, request_reply=False, silent=self._silent)
```

File: autogen/agentchat/groupchat.py [hunk 4]
```diff
@@ -671,7 +673,7 @@ async def a_run_chat(
             # Broadcast the intro
             intro = groupchat.introductions_msg()
             for agent in groupchat.agents:
-                await self.a_send(intro, agent, request_reply=False, silent=True)
+                await self.a_send(intro, agent, request_reply=False, silent=self.silent)
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:676]:
```suggestion
                await self.a_send(intro, agent, request_reply=False, silent=self._silent)
```

File: autogen/agentchat/groupchat.py [hunk 5]
```diff
@@ -689,7 +691,7 @@ async def a_run_chat(
             # broadcast the message to all agents except the speaker
             for agent in groupchat.agents:
                 if agent != speaker:
-                    await self.a_send(message, agent, request_reply=False, silent=True)
+                    await self.a_send(message, agent, request_reply=False, silent=self.silent)
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:694]:
```suggestion
                    await self.a_send(message, agent, request_reply=False, silent=self._silent)
```

File: autogen/agentchat/groupchat.py [hunk 6]
```diff
@@ -710,7 +712,7 @@ async def a_run_chat(
             if reply is None:
                 break
             # The speaker sends the message without requesting a reply
-            await speaker.a_send(reply, self, request_reply=False)
+            await speaker.a_send(reply, self, request_reply=False, silent=self.silent)
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:715]:
```suggestion
            await speaker.a_send(reply, self, request_reply=False, silent=self._silent)
```

File: autogen/agentchat/groupchat.py (hunk 2 above)
Comment (WaelKarkoub) [reply autogen/agentchat/groupchat.py:520]:
```suggestion
        silent: bool = False,
```

File: autogen/agentchat/groupchat.py [hunk 7]
```diff
@@ -516,6 +517,7 @@ def __init__(
         max_consecutive_auto_reply: Optional[int] = sys.maxsize,
         human_input_mode: Optional[str] = "NEVER",
         system_message: Optional[Union[str, List]] = "Group chat manager.",
+        silent: Optional[bool] = False,
         **kwargs,
     ):
         if (
```
Comment (WaelKarkoub) [inline autogen/agentchat/groupchat.py:523]:
```suggestion
        self._silent = silent
        if (
```

**Comment by ekzhu** [comment]
Just a reminder, you can take a look at the recent changes in conversable agent: using iostream instead of print for info output to websocket

**Comment by robraux** [comment]
@jtoy I noticed this open PR for the functionality we're both interested in. Do you have plans to wrap it up soon? I'd be happy to help finish it if you're tied up with other commitments.

**Comment by robraux** [comment]
Resolved/replaced via #2712

**Pull request closed without merging**